### ISSUE TITLE
Event callbacks not called

### ISSUE THREAD

**Issue by bitbay** [typekit/webfontloader#351]
Using custom fonts to load i just get the "loading" and the "active" events to get called.
```
...
console.log('loading fonts');
WebFont.load({
    "custom": {
        "families": families,
        "urls": urls
    },
    loading: function () {
        console.log('loading');
    },
    active: function () {
        console.log('active');
    },
    inactive: function () {
        console.log('inactive');
    },
    fontloading: function (fontFamily, fontDescription) {
        console.log('fontloading: ' + fontFamily + ' (' + fontDescription + ')');
    },
    fontactive: function (fontFamily, fontDescription) {
        console.log('fontactive: ' + fontFamily + ' (' + fontDescription + ')');
    },
    fontinactive: function (fontFamily, fontDescription) {
        console.log('fontinactive: ' + fontFamily + ' (' + fontDescription + ')');
    }
})
```

Fonts loading fine, and see the following in console:
```
loading fonts
loading
active
```

Are there any edge-cases where the other events are not "fired" or is this a bug?
Thanks.

**Comment by groundh0g** [comment]
Unless you were expecting an error, you shouldn't see the inactive events, right?

**Comment by bitbay** [comment]
@groundh0g right, in case of no errors the inactive events should not get fired/called, but as the docs state
> **fontloading - This event is triggered once for each font that's loaded.**
> **fontactive - This event is triggered once for each font that renders.**
> fontinactive - This event is triggered if the font can't be loaded.

i would expect the events in order to be (in case of _even one font_)
```
loading fonts
fontloading...
loading
fontactive...
active
```
but i am missing the "font*" events - hence my question.

**Comment by dobre-robert-marius** [comment]
+1 almost same experience for me. The only difference is that it triggers only the "loading" event, any other event is not triggered (font*, active and inactive).
Fonts are loaded correctly, classnames are set correctly, only the events are not triggered (except "loading", which works as expected).

Webfont 1.6.26, Chrome 55, windows 8.1

**Comment by JonDagata** [comment]
Having the exact same issue on some of our client sites. We had not updated or changed anything, they just simply started breaking.

**Comment by sean-mystyle** [comment]
2025 here - still seeing this issue after ~10 hours of testing and dev.  

In my case it's with a rapid-succession of font calls where a user is paginating through lists of fonts, and on each page all the fonts will load for previews, sending the WebFont.load() function repeatedly with different fonts, instead of 1 call with an array of fonts as in most the examples... For some reason, we then get an inconsistent amount of events back - usually the first one or two will throw the event back for .active(), but say out of 5 font on the page, after the first one or two the rest never have their event handler function called at all.  Sounds like the issue applies to both ways of doing it though, so I don't think the array with 1 call to WebFont.load vs 5 in a row with single fonts is the issue.

This seems to be true of both Google Fonts and Custom Fonts coming from the same server as the web app (so no CORS issues).

I've resorted to the workaround suggestion, an interval checking the html for classes, but it's still strange that the body class is being updated accurately from WebFont class doing it's thing after font loads.  If it's updating the html tag with classes accurately, you'd think it'd also be firing it's respective events.  It's almost like the event handler is overwritten or something, but then the order of course is backwards, bc then you'd expect the last in the list to handle it's even, rather than the 1st / 2nd.

I am testing in Chrome and I've tried various 1.4.7 thru 1.6.26  and found with some versions the events never fire at all, and with others there's various inconsistency, but none worked 100% of the time.

I also wonder if it's related at all to the fonts and them being on my computer etc.

In any case, the interval check for the classes is the only way I was able to get the reliable 100% recognition of google fonts and custom fonts in a reasonable response time.
